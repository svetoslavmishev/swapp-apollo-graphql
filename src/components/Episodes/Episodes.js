import React, { useContext } from 'react';
import { useQuery } from '@apollo/react-hooks';

import { Header, EpisodeCard, Loading } from '../index';
import { ALL_EPISODES } from '../../queries/queries';
import { ThemeContext } from '../../themeContext';
import styles from './EpisodesStyles';

const Episodes = () => {
  const { currentTheme } = useContext(ThemeContext);
  const classes = styles({ currentTheme });

  const { data, loading, error } = useQuery(ALL_EPISODES, {
    variables: { first: 20 }
  });

  if (loading) return <Loading />;
  if (error) return null;

  return (
    <div>
      <Header />
      <div className={classes.root}>
        {data &&
          data.allEpisodes.edges.map(episode => (
            <EpisodeCard key={episode.node.id} episode={episode.node} />
          ))}
      </div>
    </div>
  );
};

export default Episodes;
