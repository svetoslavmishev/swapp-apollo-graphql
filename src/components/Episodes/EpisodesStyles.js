import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].backgroundColor
  }
}));

export default styles;
