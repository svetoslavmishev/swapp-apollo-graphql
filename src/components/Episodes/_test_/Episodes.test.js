import React from 'react';
import { shallow } from 'enzyme';
import { ThemeContext, theme } from '../../../themeContext';
import Episodes from '../Episodes';

describe('<Episodes />', () => {
  it('renders <Episodes /> without error', () => {
    const wrapper = shallow(
      <ThemeContext.Provider value={theme}>
        <Episodes />
      </ThemeContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
