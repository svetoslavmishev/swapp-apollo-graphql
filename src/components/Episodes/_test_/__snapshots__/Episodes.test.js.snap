// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Episodes /> renders <Episodes /> without error 1`] = `ShallowWrapper {}`;
